ballerina/tests/schema-client-test.bal

```diff
@@ -0,0 +1,152 @@
+function testGetTableInfoColumnsWithConstraints() returns error? {
+    SchemaClient client1 = check new (host, user, password, "metadataDB", port);
+    TableDefinition 'table = check client1->getTableInfo("EMPLOYEES", include = sql:COLUMNS_WITH_CONSTRAINTS);
+    check client1.close();
+
+    test:assertEquals('table.name, "EMPLOYEES");
+    test:assertEquals('table.'type, "BASE TABLE");
+    test:assertEquals('table.checkConstraints, [{"name": "CHK_EmpNums", "clause": "((`EMPLOYEENUMBER` > 0) and (`REPORTSTO` > 0))"}]);
+
+    string tableCol = (<sql:ColumnDefinition[]>'table.columns).toString();
+    boolean columnCheck = tableCol.includes("EMPLOYEENUMBER") && tableCol.includes("LASTNAME") &&
+                        tableCol.includes("FIRSTNAME") && tableCol.includes("EXTENSION") &&
+                        tableCol.includes("EMAIL") && tableCol.includes("OFFICECODE") &&
+                        tableCol.includes("REPORTSTO") && tableCol.includes("JOBTITLE") &&
+                        tableCol.includes("FK_EmployeesOffice") && tableCol.includes("FK_EmployeesManager");
+
+    test:assertEquals(columnCheck, true);
+}
+
+@test:Config {
+    groups: ["metadata"]
+}
+function testGetTableInfoNegative() returns error? {
+    SchemaClient client1 = check new (host, user, password, "metadataDB", port);
+    TableDefinition|sql:Error 'table = client1->getTableInfo("EMPLOYEE", include = sql:NO_COLUMNS);
+    check client1.close();
+    if 'table is sql:Error {
+        test:assertEquals('table.message(), "The selected table does not exist or the user does not have the required privilege level to view the table.");
+    } else {
+        test:assertFail("Expected result not received");
+    }
+}
+
+@test:Config {
+    groups: ["metadata"]
+}
+function testListRoutines() returns error? {
+    SchemaClient client1 = check new (host, user, password, "metadataDB", port);
+    string[] routineList = check client1->listRoutines();
+    check client1.close();
+    test:assertEquals(routineList, ["getEmpsEmail", "getEmpsName"]);
+}
+
+@test:Config {
+    groups: ["metadata"]
+}
+function testListRoutinesNegative() returns error? {
+    SchemaClient client1 = check new (host, user, password, "metadataEmptyDB", port);
+    string[] routineList = check client1->listRoutines();
+    check client1.close();
+    test:assertEquals(routineList, []);
+}
+
+@test:Config {
+    groups: ["metadata"]
+}
+function testGetRoutineInfo() returns error? {
+    SchemaClient client1 = check new (host, user, password, "metadataDB", port);
+    sql:RoutineDefinition routine = check client1->getRoutineInfo("getEmpsName");
+    check client1.close();
+    test:assertEquals(routine.name, "getEmpsName");
+    test:assertEquals(routine.'type, "PROCEDURE");
+
+    string routineParams = (<sql:ParameterDefinition[]>routine.parameters).toString();
+    boolean paramCheck = routineParams.includes("EMPNUMBER") && routineParams.includes("FNAME");
+    test:assertEquals(paramCheck, true);
+}
```

Comment:
We should add a test case with referential constraint retrieval as well

Reply:
Referential constraints have been checked on line 86 along with check constraints
`tableCol.includes("FK_EmployeesOffice") && tableCol.includes("FK_EmployeesManager");`